### PR TITLE
dotCMS/core#26551 feat(ai-image-prompt): refactory of the ai-image-prompt

### LINKED ISSUE #26551: [UI] Review `AIImagePromptExtension` prompt extension
- Fix all the comments left by Frontend of dotCMS before merging to master (https://github.com/dotCMS/core/pull/26454)

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Refactory of the ai-image-prompt

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             
Screenshot 2023-11-27 at 11 44 04 AM

Updated
Screenshot 2023-11-27 at 11 46 04 AM

### CODE REVIEW

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.component.html [hunk 1]
```diff
@@ -1,118 +1,32 @@
...
-                        fill="#7042F0" />
-                </svg>
-            </div>
-            <span>
-                <span
-                    ><strong>Generate</strong> an AI image based on your input and requests.
-                </span>
+<ng-container *ngIf="vm$ | async as vm">
+    <p-dialog
+        [(visible)]="vm.showDialog"
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.component.html:3]: I'm curious if this is not causing unexpected behavior. 

This is a `Two-way binding` and we are passing here a value from the store. Isn't this changing the value of the store?  I know we are using the `onHide` method but that one is called after this value changes to `false`

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+     * The `generateImage` variable is a function that generates an image based on a given prompt.
+     *
+     * @param {Observable<string>} prompt$ - An observable representing the prompt.
+     * @returns {Observable} - An observable that emits the generated image.
+     */
+    readonly generateImage = this.effect((prompt$: Observable<string>) => {
+        return prompt$.pipe(
+            withLatestFrom(this.state$),
+            switchMap(([prompt, { selectedPromptType, editorContent }]) => {
+                const cleanPrompt = prompt ? prompt.trim() : '';
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts:90]:
```suggestion
                const cleanPrompt = prompt?.trim() ?? "";
```

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    /**
+     * Regenerate the image and call the generateImage effect with withLatestFrom prompt
+     */
+    readonly reGenerateContent = this.effect((trigger$: Observable<void>) => {
+        return trigger$.pipe(
+            withLatestFrom(this.state$),
+            map(([_, { prompt }]) => {
+                if (!prompt) return;
+                this.generateImage(of(prompt));
+            })
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts:124]: Shouldn't this be a `tap`? If there is a `prompt` we are not returning a value

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-input/ai-image-prompt-input.component.html [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                    fill="#7042F0" />
+                <path
+                    d="M19.6845 9.67937L16.459 11.0447C16.3233 11.1021 16.2072 11.2002 16.1254 11.3265C16.0437 11.4527 16 11.6014 16 11.7535C16 11.9056 16.0437 12.0542 16.1254 12.1805C16.2072 12.3067 16.3233 12.4048 16.459 12.4623L19.6845 13.8276L21.0001 17.175C21.0555 17.3159 21.1501 17.4364 21.2717 17.5212C21.3934 17.606 21.5366 17.6514 21.6832 17.6514C21.8297 17.6514 21.973 17.606 22.0946 17.5212C22.2163 17.4364 22.3108 17.3159 22.3662 17.175L23.6818 13.8276L26.9067 12.4623C27.0424 12.4048 27.1585 12.3067 27.2402 12.1805C27.322 12.0542 27.3656 11.9056 27.3656 11.7535C27.3656 11.6014 27.322 11.4527 27.2402 11.3265C27.1585 11.2002 27.0424 11.1021 26.9067 11.0447L23.6818 9.67937L22.3662 6.33266C22.3108 6.19183 22.2163 6.07132 22.0946 5.98648C21.973 5.90165 21.8297 5.85635 21.6832 5.85635C21.5366 5.85635 21.3934 5.90165 21.2717 5.98648C21.1501 6.07132 21.0555 6.19183 21.0001 6.33266L19.6845 9.67937Z"
+                    fill="#7042F0" />
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-input/ai-image-prompt-input.component.html:25]: Shouldn't we use an icon from PrimeNg? Maybe we can check this with Melissa

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-input/ai-image-prompt-input.component.html [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            (click)="generateImage()"
+            icon="pi pi-send"
+            label="Generate"
+            pButton
+            type="submit"></button>
+    </form>
+
+    <ng-template #autoHeaderTpl>
+        <div class="prompt-input__icon">
+            <svg
```
Comment (rjvelazco) [inline core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-input/ai-image-prompt-input.component.html:67]: Same here

**Review by rjvelazco — DISMISSED** [review]

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-input/ai-image-prompt-input.component.html (hunk 4 above)
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-input/ai-image-prompt-input.component.html:25]: I asked, and she designed that icon.

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts (hunk 2 above)
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts:90]: Changed.

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts (hunk 3 above)
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts:124]: fixed

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts [hunk 6]
```diff
@@ -50,95 +48,84 @@ export class AIImagePromptView {
...
+         * Subscription fired by the store when the dialog change of the state
+         * Handle the click of Generate button
+         */
+        this.store.isLoading$
+            .pipe(
+                filter((isLoading) => isLoading === true),
+                takeUntil(this.destroy$)
+            )
+            .subscribe(() => {
+                this.component.instance.hideDialog();
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts:92]: In the component, I use hideDialog() when a user presses ESC, clicks in the background, or clicks en X of close dialog.

[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts:92]: I will use the store in the plugin

File: core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts [hunk 7]
```diff
@@ -50,95 +48,84 @@ export class AIImagePromptView {
...
-            this.tippy?.popperInstance?.forceUpdate();
-
-            return;
-        }
-
-        if (!next.open) {
-            this.component.instance.cleanForm();
+        // show the dialog
+        if (next.open && prev.open === false) {
+            this.component.instance.showDialog(this.editor.getText());
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts:122]: done

**Review by hmoreras — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Pull request merged**